Apps/Immunization/src/Delegates/RestImmunizationDelegate.cs

```diff
@@ -41,106 +44,152 @@ namespace HealthGateway.Immunization.Delegates
     public class RestImmunizationDelegate : IImmunizationDelegate
     {
         private const string ImmunizationConfigSectionKey = "Immunization";
-
         private readonly ILogger logger;
         private readonly IHttpClientService httpClientService;
         private readonly ImmunizationConfig immunizationConfig;
 
+        /// <summary>
+        /// Gets or sets the http context accessor.
+        /// </summary>
+        private readonly IHttpContextAccessor httpContextAccessor;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="RestImmunizationDelegate"/> class.
         /// </summary>
         /// <param name="logger">Injected Logger Provider.</param>
         /// <param name="httpClientService">The injected http client service.</param>
         /// <param name="configuration">The injected configuration provider.</param>
+        /// <param name="httpContextAccessor">The Http Context accessor.</param>
         public RestImmunizationDelegate(
             ILogger<RestImmunizationDelegate> logger,
             IHttpClientService httpClientService,
-            IConfiguration configuration)
+            IConfiguration configuration,
+            IHttpContextAccessor httpContextAccessor)
         {
             this.logger = logger;
             this.httpClientService = httpClientService;
+            this.httpContextAccessor = httpContextAccessor;
             this.immunizationConfig = new ImmunizationConfig();
             configuration.Bind(ImmunizationConfigSectionKey, this.immunizationConfig);
         }
 
         private static ActivitySource Source { get; } = new ActivitySource(nameof(RestImmunizationDelegate));
 
         /// <inheritdoc/>
-        public async Task<RequestResult<PHSAResult<ImmunizationResponse>>> GetImmunizations(string bearerToken, int pageIndex = 0)
+        public async Task<RequestResult<PHSAResult<ImmunizationViewResponse>>> GetImmunization(string immunizationId)
+        {
+            using Activity? activity = Source.StartActivity("GetImmunization");
```

Comment:
whats this about?

Reply:
This is about tracing and instrumentation. When the method goes out of scope the activity is destroyed